src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -113,11 +115,17 @@ public async Task<bool> IsFullyLoadedAsync(CancellationToken cancellationToken)
 
                 var status = await GetProgressStageStatusAsync(cancellationToken).ConfigureAwait(false);
                 if (status == null)
-                {
                     return false;
-                }
 
-                return !status.IsInProgress;
+                if (status.IsInProgress)
+                    return false;
+
+                var componentModel =
+                    await _serviceProvider.GetServiceAsync<SComponentModel, IComponentModel>(throwOnFailure: false)
+                                          .WithCancellation(cancellationToken).ConfigureAwait(false);
+
+                var solutionRestoreStatusProvider = componentModel.GetService<IVsSolutionRestoreStatusProvider>();
+                return await solutionRestoreStatusProvider.IsRestoreCompleteAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
📝 We should verify with NuGet team that this is actually needed before merging it

Reply:
@emgarten Can you advise here?  We're trying to find the point that we can consider a solution reasonably "fully loaded".  i.e. that the language service has been told about all the real files/dlls necessary for normal IDE features to work properly.  The above attemps to wait until we're past the intellisense-build stage and that the nuget-restore has completed.  Do you think that's sufficient/accurate?  Thanks!

Reply:
@nkolev92 Was told you'd be good to pull in on this :)

Reply:
I think the preference would be use to Operation Progress APIs. 
cc @davkean 

The operation progress captures the fact that NuGet restore has run.

Reply:
so we are using that up on the line where we do:

```
                if (status.IsInProgress)
                    return false;
```

However, that seems to be about knowing the state of hte intellisense build.  This is more about knowing when nuget restore has completed.